### PR TITLE
Fix/ipmi status codes

### PR DESCRIPTION
**Pull request by AlirieGray**
Closes: #5755

This PR adds support for hex values in IPMI messages. It also adds unit tests for both version 1 & 2 parsing of IPMI messages.

- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by glinton** [comment]
Also, there's no need to force push on telegraf, we squash and merge so it ends up being one commit in the end.

File: plugins/inputs/ipmi_sensor/ipmi.go [hunk 1]
```diff
@@ -150,9 +150,19 @@ func parseV1(acc telegraf.Accumulator, hostname string, cmdOut []byte, measured_
 			fields["status"] = 0
 		}
 
-		if strings.Index(ipmiFields["description"], " ") > 0 {
+		description := ipmiFields["description"]
+
+		// handle hex description field
+		if description[:2] == "0x" {
```
Comment (danielnelson) [inline plugins/inputs/ipmi_sensor/ipmi.go]: Check length of description first to avoid panic

File: plugins/inputs/ipmi_sensor/ipmi.go [hunk 2]
```diff
@@ -150,9 +150,19 @@ func parseV1(acc telegraf.Accumulator, hostname string, cmdOut []byte, measured_
...
+			descriptionInt, err := strconv.ParseInt(description, 0, 64)
```
Comment (danielnelson) [inline plugins/inputs/ipmi_sensor/ipmi.go]: Check error before using descriptionInt

**Comment by safadig** [comment]
can you please update documentation for where to add hex value? Thanks

**Comment by danielnelson** [comment]
@safadig This PR should allow hex values to be parsed from the output of `ipmitool`, but it should be transparent and there is no configuration, values are reported in decimal.  Let me know if I am misunderstanding or you can open a new issue.

**Comment by safadig** [comment]
> @safadig This PR should allow hex values to be parsed from the output of `ipmitool`, but it should be transparent and there is no configuration, values are reported in decimal. Let me know if I am misunderstanding or you can open a new issue.

@danielnelson  I am trying to use the ipmi from a Dell iDRAC that has user; password, as well as a hex based encryption key to allow access. 
I can get access using ipmitool:
sudo ipmitool -I lanplus -H 192.168.1.1 -U user -P password -y 0000000000000000000000000000000000000000 sdr -v
However, this is how the documentation lists the variables in telegraf.conf. I tried inputing the encrypted hex key before root and other areas but was not successful:
`
[[inputs.ipmi_sensor]]
#   ## optionally specify one or more servers via a url matching
#   ##  [username[:password]@][protocol[(address)]]
#   ##  e.g.
#   ##    root:passwd@lan(127.0.0.1)
   servers =["root:Password@lan(192.168.1.1)"]
   interval = "30s"
   timeout = "20s"
#
#   ## Schema Version: (Optional, defaults to version 1)
#   metric_version = 2
`

**Comment by danielnelson** [comment]
Got it.  It won't be possible to pass this option right now, but I see your feature request #6858.

**Comment by Raynok** [comment]
@danielnelson, also need this feature. I believe it can be a set variable that can be commented or uncommented in the config file (similar to the privilege level variable)

encryption Key = "String"

**Pull request merged**